test/lsp.py

```diff
@@ -898,18 +899,29 @@ def main(self) -> int:
 
         return min(max(self.test_counter.failed, self.assertion_counter.failed), 127)
 
-    def setup_lsp(self, lsp: JsonRpcProcess, expose_project_root=True):
+    def setup_lsp(
+        self,
+        lsp: JsonRpcProcess,
+        expose_project_root=True,
+        analyze_all_files_in_project=False,
+        project_root_suffix = None
```

Comment:
```suggestion
        project_root_suffix=None
```

Reply:
Or with annotations:

```suggestion
        lsp: JsonRpcProcess,
        expose_project_root: bool = True,
        analyze_all_files_in_project: bool = False,
        project_root_suffix: Optional[str] = None
```

And yeah, I know it's weird to have different spacing based on whether you have an annotation or not but [that's what PEP8 actually recommends](https://peps.python.org/pep-0008/#other-recommendations). Personally, I'd be for just always having them there :)

Reply:
Also, I see that you're passing in a string for `project_root_suffix` while `self.project_root_uri` is a subclass of `pathlib.Path`. Wouldn't it be better to stick to the latter everywhere?

Reply:
@cameel I'm fine either way. I agree however, but maybe in some kind of cleanup-PR?

Reply:
About pathlib - sure. Could be a separate PR. Just noticed that and wanted to point that out.

But maybe I'm actually wrong - the other case turned out to be a string so maybe you're not really mixing them after all. Not sure.